Index: mythtv/libs/libmythtv/datadirect.cpp
===================================================================
--- mythtv/libs/libmythtv/datadirect.cpp	(revision 299141e0a69b706239beb35b096df9ed2cea8a7b)
+++ mythtv/libs/libmythtv/datadirect.cpp	(revision 4cb10e52b7e9c2ffec9111f76556a64c8fc7aba7)
@@ -545,5 +545,5 @@
     DataDirectURLs urls1(
         "Schedules Direct",
-        "http://webservices.schedulesdirect.tmsdatadirect.com"
+        "http://dd.schedulesdirect.org"
         "/schedulesdirect/tvlistings/xtvdService",
         "http://schedulesdirect.org",
