diff --git a/mythtv/libs/libmythtv/httplivestream.cpp b/mythtv/libs/libmythtv/httplivestream.cpp
index a0a47f8..7f35b0f 100644
--- a/mythtv/libs/libmythtv/httplivestream.cpp
+++ b/mythtv/libs/libmythtv/httplivestream.cpp
@@ -499,9 +499,8 @@ bool HTTPLiveStream::UpdateSizeInfo(uint16_t width, uint16_t height,
     QString newFullURL = m_httpPrefix + newOutBase + ".m3u8";
     QString newRelativeURL;
 
-    if (newFullURL.contains("/Content/GetFile"))
-        newRelativeURL = "/Content/GetFile?StorageGroup=Streaming&FileName=" +
-            newOutBase + ".m3u8";
+    if (newFullURL.contains("/StorageGroup/"))
+        newRelativeURL = "/StorageGroup/Streaming/" + newOutBase + ".m3u8";
     else
         newRelativeURL = newOutBase + ".m3u8";
 
@@ -713,14 +712,14 @@ void HTTPLiveStream::SetOutputVars(void)
     }
 
     m_httpPrefix = gCoreContext->GetSetting("HTTPLiveStreamPrefix", QString(
-        "http://%1:%2/Content/GetFile?StorageGroup=Streaming&FileName=")
+        "http://%1:%2/StorageGroup/Streaming/")
         .arg(gCoreContext->GetSetting("MasterServerIP"))
         .arg(gCoreContext->GetSetting("BackendStatusPort")));
 
     if (!gCoreContext->GetSetting("HTTPLiveStreamPrefixRel").isEmpty())
         m_httpPrefixRel = gCoreContext->GetSetting("HTTPLiveStreamPrefixRel");
-    else if (m_httpPrefix.contains("/Content/GetFile"))
-        m_httpPrefixRel = "/Content/GetFile?StorageGroup=Streaming&FileName=";
+    else if (m_httpPrefix.contains("/StorageGroup/"))
+        m_httpPrefixRel = "";
     else
         m_httpPrefixRel = "";
 }
