--- a/mythtv/programs/mythbackend/httpstatus.cpp
+++ b/mythtv/programs/mythbackend/httpstatus.cpp
@@ -467,9 +467,9 @@ void HttpStatus::FillStatusXML( QDomDocu
         QDomElement group = pDoc->createElement("Group");
 
         group.setAttribute("id"   , fsID );
-        group.setAttribute("total", (int)(iTotal>>10) );
-        group.setAttribute("used" , (int)(iUsed>>10)  );
-        group.setAttribute("free" , (int)(iAvail>>10) );
+        group.setAttribute("total", (int)(iTotal>>20) );
+        group.setAttribute("used" , (int)(iUsed>>20)  );
+        group.setAttribute("free" , (int)(iAvail>>20) );
         group.setAttribute("dir"  , directory );
 
         if (fsID == "total")
@@ -496,9 +496,9 @@ void HttpStatus::FillStatusXML( QDomDocu
             {
                 iExpirable = query.value(0).toLongLong();
             }
-            group.setAttribute("livetv", (int)(iLiveTV>>20) );
-            group.setAttribute("deleted", (int)(iDeleted>>20) );
-            group.setAttribute("expirable", (int)(iExpirable>>20) );
+            group.setAttribute("livetv", (int)(iLiveTV>>30) );
+            group.setAttribute("deleted", (int)(iDeleted>>30) );
+            group.setAttribute("expirable", (int)(iExpirable>>30) );
             total = group;
         }
         else
@@ -1269,15 +1269,15 @@ int HttpStatus::PrintMachineInfo( QTextS
                 QLocale c(QLocale::C);
 
                 os << "            <li>Total Space: ";
-                sRep = c.toString(nTotal) + " MB";
+                sRep = c.toString(nTotal) + " GB";
                 os << sRep << "</li>\r\n";
 
                 os << "            <li>Space Used: ";
-                sRep = c.toString(nUsed) + " MB";
+                sRep = c.toString(nUsed) + " GB";
                 os << sRep << "</li>\r\n";
 
                 os << "            <li>Space Free: ";
-                sRep = c.toString(nFree) + " MB";
+                sRep = c.toString(nFree) + " GB";
                 os << sRep << "</li>\r\n";
 
                 if ((nLiveTV + nDeleted + nExpirable) > 0)
@@ -1285,19 +1285,19 @@ int HttpStatus::PrintMachineInfo( QTextS
                     os << "            <li>Space Available "
                           "After Auto-expire: ";
                     sRep = c.toString(nFree + nLiveTV +
-                                      nDeleted + nExpirable) + " MB";
+                                      nDeleted + nExpirable) + " GB";
                     os << sRep << "\r\n";
                     os << "              <ul>\r\n";
                     os << "                <li>Space Used by LiveTV: ";
-                    sRep = c.toString(nLiveTV) + " MB";
+                    sRep = c.toString(nLiveTV) + " GB";
                     os << sRep << "</li>\r\n";
                     os << "                <li>Space Used by "
                           "Deleted Recordings: ";
-                    sRep = c.toString(nDeleted) + " MB";
+                    sRep = c.toString(nDeleted) + " GB";
                     os << sRep << "</li>\r\n";
                     os << "                <li>Space Used by "
                           "Auto-expirable Recordings: ";
-                    sRep = c.toString(nExpirable) + " MB";
+                    sRep = c.toString(nExpirable) + " GB";
                     os << sRep << "</li>\r\n";
                     os << "              </ul>\r\n";
                     os << "            </li>\r\n";
@@ -1353,15 +1353,15 @@ int HttpStatus::PrintMachineInfo( QTextS
                 QLocale c(QLocale::C);
 
                 os << "            <li>Total Space: ";
-                sRep = c.toString(nTotal) + " MB";
+                sRep = c.toString(nTotal) + " GB";
                 os << sRep << "</li>\r\n";
 
                 os << "            <li>Space Used: ";
-                sRep = c.toString(nUsed) + " MB";
+                sRep = c.toString(nUsed) + " GB";
                 os << sRep << "</li>\r\n";
 
                 os << "            <li>Space Free: ";
-                sRep = c.toString(nFree) + " MB";
+                sRep = c.toString(nFree) + " GB";
                 os << sRep << "</li>\r\n";
 
                 os << "          </ul>\r\n"
