diff --git a/mythtv/libs/libmythbase/bswap.h b/mythtv/libs/libmythbase/bswap.h
index ccbe27a..b608641 100644
--- a/mythtv/libs/libmythbase/bswap.h
+++ b/mythtv/libs/libmythbase/bswap.h
@@ -25,7 +25,7 @@
 #  error Byte swapping functions not defined for this platform
 #endif
 
-#ifdef bswap_32
+#if defined(bswap_32) && !defined(bswap_dbl)
 // TODO: Any reason we choose not to use bswap_64 for this?
 static __inline__ double bswap_dbl(double x)
 {
